tree_unique_args/src/subst.rs

```diff
@@ -45,6 +45,28 @@ fn var_names_available() {
     }
 }
 
+#[test]
```

Comment:
Nice test! I suggest the following for the final version
- Add a couple tests that give tuple types
- Make the run_test method check that your typechecker gives the program a type (to ensure you are not missing rules)
- Write a rule that panics if something gets two different types